zeebe/gateway/src/main/java/io/camunda/zeebe/gateway/impl/broker/request/BrokerUpdateJobRequest.java

```diff
@@ -12,17 +12,23 @@
 import io.camunda.zeebe.protocol.record.ValueType;
 import io.camunda.zeebe.protocol.record.intent.JobIntent;
 import io.camunda.zeebe.util.buffer.BufferWriter;
+import java.util.Map;
 import org.agrona.DirectBuffer;
 
 public class BrokerUpdateJobRequest extends BrokerExecuteCommand<JobRecord> {
 
   private final JobRecord requestDto = new JobRecord();
 
-  public BrokerUpdateJobRequest(final long jobKey, final int retries, final long timeout) {
+  public BrokerUpdateJobRequest(final long jobKey, final Map<String, Number> changeset) {
     super(ValueType.JOB, JobIntent.UPDATE);
     request.setKey(jobKey);
-    requestDto.setRetries(retries);
-    requestDto.setTimeout(timeout);
+    requestDto.setChangedAttributes(changeset.keySet());
+    if (changeset.get(JobRecord.RETRIES) != null) {
+      requestDto.setRetries(changeset.get(JobRecord.RETRIES).intValue());
+    }
+    if (changeset.get(JobRecord.TIMEOUT) != null) {
+      requestDto.setTimeout(changeset.get(JobRecord.TIMEOUT).longValue());
+    }
```

Comment:
💭 I now remember why I also chose to use the `UserTaskRecord` as the "changeset" for the broker request; it basically is a type-safe map instead of an "anonymous" map like we have here. 

We cannot rely on the data types we receive here in `changeset`. This broker request constructor here might have to check that the data is an Integer or Long before simply using it as such. If you have the JobRecord here, it can only contain the correct types. You can then rely on it here.

Reply:
This refers to the test case if described earlier with
> Conversion of the changed attribute value is failing (we accept any Number types currently, I think, we can probably make that fail in a test if we want to to see if our processor can handle that 🙂)

Reply:
I'm not a big fan of using those records outside the engine honestly. I don't know what you think about this, but I'd prefer leave it like this (we are confident about data type because we get them from the request)

Reply:
Yeah, I generally understand the point with the record but we already use the record in those broker request classes heavily already anyway 😄 
Regarding confidence, yes, currently this broker request class is only used by our REST gateway. But you know how it is, it doesn't always stay like that. The broker request doesn't know anything about its callers and it is now making assumptions on what they will provide. I simply think that is a bit dangerous and I don't want to look into an incident later trying to find out why we are running into weird JDK exceptions 🤣

Reply:
> Yeah, I generally understand the point with the record but we already use the record in those broker request classes heavily already anyway 😄

Yes, this should be the only place where we use them outside the engine.

But I wanto to find a solution that works for both of us, so what you suggest to do ? 😄

Reply:
If we use the JobRecord as the changeset, we'll also use it more heavily in the REST controller to "fill it with data", right? I see your concern, might be something we also want to change for the user task record then as a follow-up so it's consistent. I totally understand this concern and I agree that keeping the records as close to the broker classes as possible only and don't leak them too much elsewhere 👍 

Looking at the issue, I see two ways currently:
- Introduce a proper DTO with specific types for the changeset attributes instead of a semi-anonymous Java `Map` with target type `Number`.
- Do validation in the broker request class for the changeset attributes.

Out of those two, I would slightly prefer the first one, I think. It relieves the broker request class from checking its inputs and puts the responsibility on the caller with an explicit contract (retries and timeout need to be proper Long or Integer values).

Reply:
I'll go for the first one 👍  thanks